ml-package-versions.yml

```diff
@@ -0,0 +1,151 @@
+sklearn:
+  package_info:
+    pip_release: "scikit-learn"
+    install_dev: |
+      pip install git+https://github.com/scikit-learn/scikit-learn.git@master
+
+  models:
+    minimum: "0.20.4"
+    maximum: "0.23.2"
+    run: |
+      pytest tests/sklearn/test_sklearn_model_export.py --large
+
+  autologging:
+    minimum: "0.20.4"
+    maximum: "0.23.2"
+    requirements: ["matplotlib"]
+    run: |
+      pytest tests/sklearn/test_sklearn_training_session.py --large
+      pytest tests/sklearn/test_sklearn_autolog.py --large
+
+pytorch:
+  package_info:
+    pip_release: "torch"
+    install_dev: |
+      pip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+
+  models:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements: ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+  autologging:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements: ["torchvision", "pytorch-lightning", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_autolog.py --large
+
+tensorflow-1.x:
+  package_info:
+    pip_release: "tensorflow"
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow/test_tensorflow_model_export.py --large
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow_autolog/test_tensorflow_autolog.py --large
+
+tensorflow-2.x:
+  package_info:
+    pip_release: "tensorflow"
+    install_dev: |
+      pip install tf-nightly
+
+  models:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    run: |
+      pytest tests/tensorflow/test_tensorflow2_model_export.py --large
+
+  autologging:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    requirements:
+      "< 2.2": ["h5py<3.0"]
+    run: |
+      pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+
+keras:
+  package_info:
+    pip_release: "keras"
+    install_dev: |
+      # In keras >= 2.4.0, `keras` simply redirects to `tensorflow.keras`:
+      # https://github.com/keras-team/keras#multi-backend-keras-and-tfkeras
+      pip install tf-nightly 'keras>=2.4.0'
+
+  models:
+    minimum: "2.0.0"
+    maximum: "2.4.3"
+    requirements:
+      "< 2.3.1": ["tensorflow==1.15.4", "h5py<3.0", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+      "== 2.3.1": ["tensorflow==2.2.1", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
```

Comment:
Was there a particular reason that Keras `2.3.1` depends on exactly TF `2.2.1`? (Sorry if we already discussed this). If so, can we add an inline comment explaining why?